pkg/lib/cockpit-components-modifications.jsx

```diff
@@ -37,85 +37,77 @@
                     .then(() => {
-                        this.setState({ copied: true });
-                        this.timeoutId = setTimeout(() => {
-                            this.setState({ copied: false });
-                        }, 3000);
+                        setCopied(true);
+                        setTimeoutId(setTimeout(() => {
+                            setCopied(false);
+                            setTimeoutId(null);
+                        }, 3000));
                     })
                     .catch(e => console.error('Text could not be copied: ', e ? e.toString() : ""));
         } catch (error) {
             console.error('Text could not be copied: ', error.toString());
         }
-    }
+    };
 
-    render() {
-        const footer = (
-            <>
-                <Button variant='secondary' className="btn-clipboard" onClick={this.copyToClipboard} icon={this.state.copied ? <CheckIcon className="green-icon" /> : <CopyIcon />}>
-                    { _("Copy to clipboard") }
-                </Button>
-                <Button variant='secondary' className='btn-cancel' onClick={this.props.onClose}>
-                    { _("Close") }
-                </Button>
-            </>
-        );
-        return (
-            <Modal isOpen={this.props.show} className="automation-script-modal"
-                   position="top" variant="medium"
-                   onClose={this.props.onClose}
-                   footer={footer}
-                   title={_("Automation script") }>
-                <Tabs activeKey={this.state.active_tab} onSelect={this.handleSelect}>
-                    <Tab eventKey="shell" title={_("Shell script")}>
-                        <TextArea resizeOrientation='vertical' readOnlyVariant="default" defaultValue={this.props.shell.trim()} />
-                    </Tab>
-                    <Tab eventKey="ansible" title={_("Ansible")}>
-                        <TextArea resizeOrientation='vertical' readOnlyVariant="default" defaultValue={this.props.ansible.trim()} />
-                        <div className="ansible-docs-link">
-                            <OutlinedQuestionCircleIcon />
-                            { _("Create new task file with this content.") }
-                            <Button variant="link" component="a" href="https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_reuse_roles.html"
-                                    target="_blank" rel="noopener noreferrer"
-                                    icon={<ExternalLinkAltIcon />}>
-                                { _("Ansible roles documentation") }
-                            </Button>
-                        </div>
-                    </Tab>
-                </Tabs>
-            </Modal>
-        );
-    }
-}
+    const footer = (
+        <>
+            <Button variant='secondary' className="btn-clipboard" onClick={copyToClipboard} icon={copied ? <CheckIcon className="green-icon" /> : <CopyIcon />}>
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19370-20230922-100423-663bbd9c-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)